packages/components/src/site-thumbnail/use-mshots-img.tsx

```diff
@@ -24,98 +23,55 @@ export type MShotsOptions = {
 	requeue?: boolean;
 };
 
-// This custom react hook returns undefined while the image is loading and
-// a HTMLImageElement (i.e. the class you get from `new Image()`) once loading
-// is complete.
-//
-// It also triggers a re-render (via setState()) when the value changes, so just
-// check if it's truthy and then treat it like any other Image.
-//
-// Note the loading may occur immediately and synchronously if the image is
-// already or may take up to several seconds if mshots has to generate and cache
-// new images.
-//
-// The calling code doesn't need to worry about the details except that you'll
-// want some sort of loading display.
-//
-// Inspired by https://stackoverflow.com/a/60458593
 export const useMshotsImg = (
 	src: string,
 	options: MShotsOptions
-): HTMLImageElement | undefined => {
-	const [ loadedImg, setLoadedImg ] = useState< HTMLImageElement >();
-	const [ count, setCount ] = useState( 0 );
-	const previousSrc = useRef( src );
-
-	const imgRef = useRef< HTMLImageElement >();
-	const timeoutIdRef = useRef< number >();
-
-	const previousImg = useRef< HTMLImageElement >();
-	const previousOptions = useRef< MShotsOptions >();
-	// Oddly, we need to assign to current here after ref creation in order to
-	// pass the equivalence check and avoid a spurious reset
-	previousOptions.current = options;
+): {
+	src: string;
+	isLoading: boolean;
+	isError: boolean;
+	imgRef: React.MutableRefObject< HTMLImageElement | null >;
+} => {
+	const [ retryCount, setRetryCount ] = useState( 0 );
```

Comment:
The retry count is meant to keep refreshing the "same" mshot url and update this default image after each retry.
![default](https://user-images.githubusercontent.com/779993/181082468-317dafd7-42f4-4d7f-a039-b8d6a8df8f67.gif)